configserver/src/main/java/com/yahoo/vespa/config/server/session/RemoteSession.java

```diff
@@ -135,4 +139,34 @@ public void delete() {
         transaction.close();
     }
 
+    public ApplicationId getApplicationId() { return zooKeeperClient.readApplicationId(); }
+
+    public Optional<DockerImage> getDockerImageRepository() { return zooKeeperClient.readDockerImageRepository(); }
+
+    public Version getVespaVersion() { return zooKeeperClient.readVespaVersion(); }
+
+    public Optional<AthenzDomain> getAthenzDomain() { return zooKeeperClient.readAthenzDomain(); }
+
+    public AllocatedHosts getAllocatedHosts() {
+        return zooKeeperClient.getAllocatedHosts();
+    }
+
+    // Note: Assumes monotonically increasing session ids
+    public boolean isNewerThan(long sessionId) {
+        return getSessionId() > sessionId;
+    }
+
+    public Transaction createDeactivateTransaction() {
+        return createSetStatusTransaction(Status.DEACTIVATE);
+    }
+
+    private Transaction createSetStatusTransaction(Status status) {
+        return zooKeeperClient.createWriteStatusTransaction(status);
+    }
+
+    public ApplicationMetaData getMetaData() {
+        return zooKeeperClient.loadApplicationPackage().getMetaData();
+    }
```

Comment:
Could probably remove most of these from LocalSession, if they're not to be used on that?

Reply:
I think I removed those that could be removed. I want to do more refactoring in this area, but let's try this first.